components/Login/SignUp.tsx

```diff
@@ -90,18 +96,27 @@ function SignUp({ navigation, route }: SignUpProps) {
 
   const submit = async () => {
     if (validateAll()) {
+      // Create user through authentication
       try {
-        const { user } = await Auth.signUp({
+        await Auth.signUp({
           username: email,
           password,
           attributes: {
             name,
             email,
           },
         });
-        console.log(user);
+
+        // Create user through datastore
+        const input = { email, name };
+        const res = (await API.graphql(
+          graphqlOperation(createUser, { input })
+        )) as { data: CreateUserMutation };
+        dispatch(setUser(res?.data?.createUser));
+
         navigation.navigate("CreateAccountCode", { email });
-      } catch {
+      } catch (error) {
+        console.error(error);
```

Comment:
If possible we could maybe add a small error text saying something like "Sorry we couldn't sign you up for an account at this time." idk only if you want to

Reply:
Good idea! I added error messages if the password is too short or the email was already taken by another user